src/anemoi/models/models/encoder_processor_decoder.py

```diff
@@ -239,26 +245,29 @@ def forward(self, x: Tensor, model_comm_group: Optional[ProcessGroup] = None) ->
         # add skip connection (hidden -> hidden)
         x_latent_proc = x_latent_proc + x_latent
 
-        # Run decoder
-        x_out = self._run_mapper(
-            self.decoder,
-            (x_latent_proc, x_data_latent),
-            batch_size=batch_size,
-            shard_shapes=(shard_shapes_hidden, shard_shapes_data),
-            model_comm_group=model_comm_group,
-        )
+        # Run decoders
+        x_out = {}
+        for data, decoder in self.decoders.items():
+            x_out[data] = self._run_mapper(
+                decoder,
+                (x_latent_proc, x_data_latent[data]),
+                batch_size=batch_size,
+                shard_shapes=(shard_shapes_hidden, shard_shapes_data[data]),
+                model_comm_group=model_comm_group,
+            )
 
-        x_out = (
-            einops.rearrange(
-                x_out,
-                "(batch ensemble grid) vars -> batch ensemble grid vars",
-                batch=batch_size,
-                ensemble=ensemble_size,
+            x_out[data] = (
+                einops.rearrange(
+                    x_out[data],
+                    "(batch ensemble grid) vars -> batch ensemble grid vars",
+                    batch=batch_size,
+                    ensemble=ensemble_size,
+                )
+                .to(dtype=x.dtype)
+                .clone()
             )
-            .to(dtype=x.dtype)
-            .clone()
-        )
 
-        # residual connection (just for the prognostic variables)
-        x_out[..., self._internal_output_idx] += x[:, -1, :, :, self._internal_input_idx]
-        return x_out
+            # residual connection (just for the prognostic variables)
```

Comment:
The comment explains what happens in the if (and should be moved there), but not what the if itself is for, limiting shortcut to names that are also in the encoders.